client/dashboard/task-list/tasks/shipping/index.js

```diff
@@ -0,0 +1,211 @@
+
+class Shipping extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initialState = {
+			isPending: false,
+			step: 'store_location',
+			shippingZones: [],
+		};
+
+		this.state = this.initialState;
+
+		this.completeStep = this.completeStep.bind( this );
+	}
+
+	componentDidMount() {
+		this.reset();
+	}
+
+	reset() {
+		this.setState( this.initialState );
+	}
+
+	async fetchShippingZones() {
+		this.setState( { isPending: true } );
+		const { countryCode, countryName } = this.props;
+		const shippingZones = [];
+		const zones = await apiFetch( { path: '/wc/v3/shipping/zones' } );
+		let hasCountryZone = false;
+
+		await Promise.all(
+			zones.map( async zone => {
+				// "Rest of the world zone"
+				if ( 0 === zone.id ) {
+					zone.methods = await apiFetch( { path: `/wc/v3/shipping/zones/${ zone.id }/methods` } );
+					zone.name = __( 'Rest of the world', 'woocommerce-admin' );
+					zone.toggleEnabled = true;
+					shippingZones.push( zone );
+					return;
+				}
+
+				// Return any zone with a single location matching the country zone.
+				zone.locations = await apiFetch( { path: `/wc/v3/shipping/zones/${ zone.id }/locations` } );
+				if ( 1 === zone.locations.length && countryCode === zone.locations[ 0 ].code ) {
+					zone.methods = await apiFetch( { path: `/wc/v3/shipping/zones/${ zone.id }/methods` } );
+					shippingZones.push( zone );
+					hasCountryZone = true;
+				}
+			} )
+		);
+
+		// Create the default store country zone if it doesn't exist.
+		if ( ! hasCountryZone ) {
+			const zone = await apiFetch( {
+				method: 'POST',
+				path: '/wc/v3/shipping/zones',
+				data: { name: countryName },
+			} );
+			zone.locations = await apiFetch( {
+				method: 'POST',
+				path: `/wc/v3/shipping/zones/${ zone.id }/locations`,
+				data: [ { code: countryCode, type: 'country' } ],
+			} );
+			shippingZones.push( zone );
+		}
+
+		shippingZones.reverse();
+
+		this.setState( { isPending: false, shippingZones } );
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		const { countryCode, settings } = this.props;
+		const {
+			woocommerce_store_address,
+			woocommerce_default_country,
+			woocommerce_store_postcode,
+		} = settings;
+		const { step } = this.state;
+
+		if (
+			'store_location' === step &&
+			woocommerce_store_address &&
+			woocommerce_default_country &&
+			woocommerce_store_postcode
+		) {
+			this.completeStep();
+		}
+
+		if (
+			prevProps.countryCode !== countryCode ||
+			( 'rates' === step && 'rates' !== prevState.step )
+		) {
+			this.fetchShippingZones();
+		}
+	}
+
+	completeStep() {
+		const { step } = this.state;
+		const steps = this.getSteps();
+		const currentStepIndex = steps.findIndex( s => s.key === step );
+		const nextStep = steps[ currentStepIndex + 1 ];
+
+		if ( nextStep ) {
+			this.setState( { step: nextStep.key } );
+		} else {
+			this.setState( { step: steps[ 0 ].key } );
+			getHistory().push( getNewPath( '/', {}, {} ) );
```

Comment:
This seems to be adding `0=%2F` to my query string.

Reply:
Nice catch!  Wrong param order here.